src/Microsoft.Azure.SignalR.Management/RestHubLifetimeManager.cs

```diff
@@ -92,31 +123,119 @@ public override Task SendUsersAsync(IReadOnlyList<string> userIds, string method
-        }
+            if (String.IsNullOrEmpty(userId))
+            {
+                throw new ArgumentException(NullOrEmptyStringErrorMessage, nameof(userId));
+            }
 
-        private Task CallRestApi(RestApiEndpoint endpoint, HttpMethod httpMethod, string methodName = null, object[] args = null, CancellationToken cancellationToken = default)
-        {
-            var payload = httpMethod == HttpMethod.Post ? new PayloadMessage { Target = methodName, Arguments = args } : null;
-            var httpClient = HttpClientFactory.CreateClient();
-            var request = GenerateHttpRequest(endpoint.Audience, payload, endpoint.Token, HttpMethod.Post);
-            return httpClient.SendAsync(request);
+            if (String.IsNullOrEmpty(groupName))
+            {
+                throw new ArgumentException(NullOrEmptyStringErrorMessage, nameof(groupName));
+            }
+
+            var api = _restApiProvider.GetUserGroupManagementEndpoint(userId, groupName);
+            var request = BuildRequest(api, HttpMethod.Delete, null, null, cancellationToken);
+            return CallRestApiAsync(request);
         }
 
-        private HttpRequestMessage GenerateHttpRequest(string url, PayloadMessage payload, string tokenString, HttpMethod httpMethod)
+        private static HttpRequestMessage GenerateHttpRequest(string url, PayloadMessage payload, string tokenString, HttpMethod httpMethod)
         {
             var request = new HttpRequestMessage(httpMethod, url);
             request.Headers.Authorization = new AuthenticationHeaderValue("Bearer", tokenString);
             request.Headers.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
             request.Content = new StringContent(JsonConvert.SerializeObject(payload), Encoding.UTF8, "application/json");
             return request;
         }
+
+        private static void ThrowException(Exception innerException, HttpStatusCode? statusCode, string requestUri, string detail = null)
+        {
+            switch (statusCode)
+            {
+                case HttpStatusCode.BadRequest:
+                    {
+                        throw new AzureSignalRBadRequestException(requestUri, innerException, detail);
+                    }
+                case HttpStatusCode.Unauthorized:
+                    {
+                        throw new AzureSignalRUnauthorizedException(requestUri, innerException);
+                    }
+                case HttpStatusCode.NotFound:
+                    {
+                        throw new AzureSignalRUnableToAccessException(requestUri, innerException);
+                    }
+                default:
+                    {
+                        throw new AzureSignalRRuntimeException(requestUri, innerException);
+                    }
+            }
+        }
+
+        private static void ThrowException(Exception innerException, string requestUri, string detail = null)
```

Comment:
This is confusing, what's the difference between the two ThrowExceptions? The name of method should be more self-explain.